nemo/collections/asr/modules/conformer_modules.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import torch.nn as nn
+
+from nemo.collections.asr.modules.activations import Swish
+
+__all__ = ['ConformerConvolution', 'ConformerFeedForward']
+
+
+class ConformerConvolution(nn.Module):
+    """The convolution module for the Conformer model.
+    Args:
+        d_model (int): hidden dimension
+        kernel_size (int): kernel size for depthwise convolution
+    """
+
+    def __init__(self, d_model, kernel_size):
+        super(ConformerConvolution, self).__init__()
+        assert (kernel_size - 1) % 2 == 0
+        self.d_model = d_model
+
+        self.pointwise_conv1 = nn.Conv1d(
+            in_channels=d_model, out_channels=d_model * 2, kernel_size=1, stride=1, padding=0, bias=True
+        )
+        self.depthwise_conv = nn.Conv1d(
+            in_channels=d_model,
+            out_channels=d_model,
+            kernel_size=kernel_size,
+            stride=1,
+            padding=(kernel_size - 1) // 2,
+            groups=d_model,
+            bias=True,
+        )
+        self.batch_norm = nn.BatchNorm1d(d_model)
+        self.activation = Swish()
+        self.pointwise_conv2 = nn.Conv1d(
+            in_channels=d_model, out_channels=d_model, kernel_size=1, stride=1, padding=0, bias=True
+        )
+
+    def forward(self, x):
+        x = x.transpose(1, 2)
+        x = self.pointwise_conv1(x)
+
+        x = nn.functional.glu(x, dim=1)
+
+        x = self.depthwise_conv(x)
+
+        x = self.batch_norm(x)
+        x = self.activation(x)
+
+        x = self.pointwise_conv2(x)
+        x = x.transpose(1, 2)
+        return x
+
+
+class ConformerFeedForward(nn.Module):
```

Comment:
I would urge you to subclass NeuralModule and add type checking. Alternatively, please move this to parts as opposed to keeping it inside the modules directory.

Reply:
Moved.